src/main/java/com/powsybl/openloadflow/network/impl/LfBranchImpl.java

```diff
@@ -229,37 +228,70 @@ public List<BranchResult> createBranchResult(double preContingencyBranchP1, doub
+                return getLimits1(type, branch.getCurrentLimits1().orElse(null), limitReductionManager);
             case VOLTAGE:
             default:
                 throw new UnsupportedOperationException(String.format("Getting %s limits is not supported.", type.name()));
         }
     }
 
     @Override
-    public List<LfLimit> getLimits2(final LimitType type) {
+    public List<LfLimit> getLimits2(final LimitType type, LimitReductionManager limitReductionManager) {
         var branch = getBranch();
         switch (type) {
             case ACTIVE_POWER:
-                return getLimits2(type, branch.getActivePowerLimits2().orElse(null));
+                return getLimits2(type, branch.getActivePowerLimits2().orElse(null), limitReductionManager);
             case APPARENT_POWER:
-                return getLimits2(type, branch.getApparentPowerLimits2().orElse(null));
+                return getLimits2(type, branch.getApparentPowerLimits2().orElse(null), limitReductionManager);
             case CURRENT:
-                return getLimits2(type, branch.getCurrentLimits2().orElse(null));
+                return getLimits2(type, branch.getCurrentLimits2().orElse(null), limitReductionManager);
             case VOLTAGE:
             default:
                 throw new UnsupportedOperationException(String.format("Getting %s limits is not supported.", type.name()));
         }
     }
 
+    @Override
+    public List<Double> getLimitReductions(TwoSides side, LimitReductionManager limitReductionManager, LoadingLimits limits) {
+        if (limits == null) {
+            return Collections.emptyList();
+        }
+        if (limits.getLimitType() != LimitType.CURRENT) {
+            return Collections.emptyList();
+        }
+        if (limitReductionManager == null || limitReductionManager.isEmpty()) {
+            return Collections.emptyList();
+        }
+        Double[] limitReductions = new Double[limits.getTemporaryLimits().size() + 1];
+        Arrays.fill(limitReductions, 1.);
+        double nominalV = branchRef.get().getTerminal(side).getVoltageLevel().getNominalV();
+        for (LimitReductionManager.TerminalLimitReduction terminalLimitReduction : limitReductionManager.getTerminalLimitReductions()) {
+            if (terminalLimitReduction.getNominalV().contains(nominalV)) {
+                if (terminalLimitReduction.isPermanent()) {
+                    limitReductions[0] = terminalLimitReduction.getReduction();
+                }
+                if (terminalLimitReduction.getAcceptableDuration() != null) {
+                    int i = 1;
+                    for (LoadingLimits.TemporaryLimit temporaryLimit : limits.getTemporaryLimits()) {
+                        if (terminalLimitReduction.getAcceptableDuration().contains(temporaryLimit.getAcceptableDuration())) {
+                            limitReductions[i] = terminalLimitReduction.getReduction();
+                        }
```

Comment:
1. The array is initialized with (_temporary limits number_ +1) values [^1] set to  `1.` (see l.273 and 274). Even if no reduction is defined for temporary limits, there will always be a reduction value for each temporary limit (`1.`);
2. That's it: when a limit matches the criteria of several `LimitReduction`s, the last one applies. I will add a comment and a unit test to clearly identify (and check) this point.

[^1]: index 0 is reserved for the permanent limit

Reply:
Thanks @olperr1 . I get it (although it is a bit hard to read )
What about the situation where several limits match the duration and voltage ?
It seems in this implementation that the last limit overrides the other.
Is it really the desired behaviour ? Won't it be difficult to troubleshoot is in a complex software several limit reductions would apply for a line ?

Reply:
One more thing. This algorithm is quite expensive since you need to iterate on all limit reductions.
And the function might be called quite frequently (for example for at least once per contingency and per branch).

Why not at least caching the result for each LfBranch so that the iteration is  made at most once per branch ?